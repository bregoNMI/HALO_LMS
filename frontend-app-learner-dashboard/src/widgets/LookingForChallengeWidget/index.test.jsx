import { shallow } from '@edx/react-unit-test-utils';

import LookingForChallengeWidget from '.';

jest.mock('hooks', () => ({
  reduxHooks: {
    usePlatformSettingsData: () => ({
      courseSearchUrl: 'http://localhost:18000/course-search-url',
    }),
  },
}));

jest.mock('../RecommendationsPanel/track', () => ({
  findCoursesWidgetClicked: (href) => jest.fn().mockName(`track.findCoursesWidgetClicked('${href}')`),
}));

describe('LookingForChallengeWidget', () => {
  describe('snapshots', () => {
    test('default', () => {
      const wrapper = shallow(<LookingForChallengeWidget />);
      expect(wrapper.snapshot).toMatchSnapshot();
    });
  });
});
