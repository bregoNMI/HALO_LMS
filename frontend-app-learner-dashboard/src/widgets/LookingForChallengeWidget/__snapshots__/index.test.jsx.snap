// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LookingForChallengeWidget snapshots default 1`] = `
<Card
  id="looking-for-challenge-widget"
  orientation="horizontal"
>
  <Card.ImageCap
    src="icon/mock/path"
    srcAlt="course side widget"
  />
  <Card.Body
    className="m-auto pr-2"
  >
    <h4>
      Looking for a new challenge?
    </h4>
    <h5>
      <Hyperlink
        className="d-flex align-items-center"
        destination="http://localhost:18000/course-search-url"
        onClick={[MockFunction track.findCoursesWidgetClicked('http://localhost:18000/course-search-url')]}
        variant="brand"
      >
        <format-message-function
          message={
            Object {
              "defaultMessage": "Find a course {arrow}",
              "description": "Button to explore more courses",
              "id": "WidgetSidebar.findCoursesButton",
            }
          }
          values={
            Object {
              "arrow": <Icon
                className="mx-1"
              />,
            }
          }
        />
      </Hyperlink>
    </h5>
  </Card.Body>
</Card>
`;
